test/e2e/framework/pod/resource.go

```diff
@@ -572,3 +575,129 @@ func IsPodActive(p *v1.Pod) bool {
 		v1.PodFailed != p.Status.Phase &&
 		p.DeletionTimestamp == nil
 }
+
+const (
+	CPUPeriod             string = "100000"
+	CgroupCPUPeriod       string = "/sys/fs/cgroup/cpu/cpu.cfs_period_us"
+	CgroupCPUShares       string = "/sys/fs/cgroup/cpu/cpu.shares"
+	CgroupCPUQuota        string = "/sys/fs/cgroup/cpu/cpu.cfs_quota_us"
+	CgroupMemLimit        string = "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+	Cgroupv2MemLimit      string = "/sys/fs/cgroup/memory.max"
+	Cgroupv2MemRequest    string = "/sys/fs/cgroup/memory.min"
+	Cgroupv2CPULimit      string = "/sys/fs/cgroup/cpu.max"
+	Cgroupv2CPURequest    string = "/sys/fs/cgroup/cpu.weight"
+	cgroupFsPath          string = "/sys/fs/cgroup"
+	cgroupv2CPUWeightFile string = "cpu.weight"
+	cgroupv2CPULimitFile  string = "cpu.max"
+	cgroupv2MemLimitFile  string = "memory.max"
+	cgroupVolumeName      string = "sysfscgroup"
+	cgroupMountPath       string = "/sysfscgroup"
+)
+
+type ContainerResources struct {
+	CPUReq              string
+	CPULim              string
+	MemReq              string
+	MemLim              string
+	EphStorReq          string
+	EphStorLim          string
+	ExtendedResourceReq string
+	ExtendedResourceLim string
+}
+
+func (cr *ContainerResources) ResourceRequirements() *v1.ResourceRequirements {
+	if cr == nil {
+		return nil
+	}
+
+	var lim, req v1.ResourceList
+	if cr.CPULim != "" || cr.MemLim != "" || cr.EphStorLim != "" {
+		lim = make(v1.ResourceList)
+	}
+	if cr.CPUReq != "" || cr.MemReq != "" || cr.EphStorReq != "" {
+		req = make(v1.ResourceList)
+	}
+	if cr.CPULim != "" {
+		lim[v1.ResourceCPU] = resource.MustParse(cr.CPULim)
+	}
+	if cr.MemLim != "" {
+		lim[v1.ResourceMemory] = resource.MustParse(cr.MemLim)
+	}
+	if cr.EphStorLim != "" {
+		lim[v1.ResourceEphemeralStorage] = resource.MustParse(cr.EphStorLim)
+	}
+	if cr.CPUReq != "" {
+		req[v1.ResourceCPU] = resource.MustParse(cr.CPUReq)
+	}
+	if cr.MemReq != "" {
+		req[v1.ResourceMemory] = resource.MustParse(cr.MemReq)
+	}
+	if cr.EphStorReq != "" {
+		req[v1.ResourceEphemeralStorage] = resource.MustParse(cr.EphStorReq)
+	}
+	return &v1.ResourceRequirements{Limits: lim, Requests: req}
+}
+
+func CreateHostPathVolumeForCgroup() v1.Volume {
+	return v1.Volume{
+		Name: cgroupVolumeName,
+		VolumeSource: v1.VolumeSource{
+			HostPath: &v1.HostPathVolumeSource{Path: cgroupFsPath},
+		},
+	}
+}
+
+func CreateVolumeMountForCgroup() v1.VolumeMount {
+	return v1.VolumeMount{
+		Name:      cgroupVolumeName,
+		MountPath: cgroupMountPath,
+	}
+}
+
+// VerifyPodCgroups verifies pod cgroup is configured on a node as expected.
+func VerifyPodCgroups(ctx context.Context, f *framework.Framework, pod *v1.Pod, info *ContainerResources) error {
```

Comment:
Same feedback here. It would be good to create smaller functions to verify each cgroup value.

```
func VerifyPodCgroups(ctx context.Context, f *framework.Framework, pod *v1.Pod, info *ContainerResources) error {
	ginkgo.GinkgoHelper()

	// Extract pod cgroup directory path
	podCgPath, err := getPodCgroupPath(f, pod)
	if err != nil {
		return err
	}

	// Verify cgroup values
	expectedResources := info.ResourceRequirements()
	var errs []error
	errs = append(errs, verifyCPUWeight(f, pod, podCgPath, expectedResources))
	errs = append(errs, verifyCPULimit(f, pod, podCgPath, expectedResources))
	errs = append(errs, verifyMemoryLimit(f, pod, podCgPath, expectedResources))

	return utilerrors.NewAggregate(errs)
}
```